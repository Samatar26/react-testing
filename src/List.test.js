import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import List from './List';

describe('List', () => {
  it('Should snapshot list', () => {
    const items = ['first', 'second'];
    const wrapper = shallow(<List items={items} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Should test item props as undefined', () => {
    const wrapper = shallow(<List />);
  });

  it('Should render the right text', () => {
    const items = ['Hello world'];
    const wrapper = shallow(<List items={items} />);
    expect(wrapper.find('li').first().text()).toBe(items[0]);
  });
});
