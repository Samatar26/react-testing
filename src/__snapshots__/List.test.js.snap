exports[`List Should snapshot list 1`] = `
<ul>
  <li>
    first
  </li>
  <li>
    second
  </li>
</ul>
`;
